sdc/datatypes/hpat_pandas_series_functions.py

```diff
@@ -1962,10 +1963,31 @@ def hpat_pandas_series_isin(self, values):
     if not isinstance(values, (types.Set, types.List)):
         ty_checker.raise_exc(values, 'set or list', 'values')
 
-    def hpat_pandas_series_isin_impl(self, values):
-        # TODO: replace with below line when Numba supports np.isin in nopython mode
-        # return pandas.Series(np.isin(self._data, values))
-        return pandas.Series(data=[(x in values) for x in self._data], index=self._index, name=self._name)
+    if isinstance(values[0], (types.UnicodeType, types.StringLiteral)):
```

Comment:
Hm, values[0] doesn't make sense for a Set (not at runtime, not at compile time), it's really strange that this works at all. Probably it should be:
values.dtype
```suggestion
    if isinstance(values.dtype, (types.UnicodeType, types.StringLiteral)):
```